src/services/catalog.service.ts

```diff
@@ -35,8 +38,46 @@ export class CatalogService {
         }
       });  // .ea category
     });  // .ea tag
+    if (_.isEmpty(catsBySubCats)) {
+      catsBySubCats[otherId] = otherId;
+    }
     return catsBySubCats;
   }
+
+  public hasCategory(item: any, category: string) {
+    return _.includes(item.catsBySubCats, category);
+  }
+
+  public hasSubCategory(item: any, subCategory: string) {
+    return _.includes(Object.keys(item.catsBySubCats), subCategory);
+  }
+
+  /**
+   * Return a new Array of only those Categories and SubCategories which
+   * exist in the passed in items.
+   * @param items
+   * @returns {Array}
+   */
+  public removeEmptyCategories(items: IServiceItem) {
```

Comment:
> Instead it's probably easier just to have an ng-if not empty check in the ng-repeat in the view.
@spadgett This might be getting back to what you said earlier about that console creates a map initially.  At the moment in catalog we don't know if a category and/or sub-category is 'empty' up front.